src/lib/registration/ui/migration_repos_workflow.rb

```diff
@@ -483,6 +533,23 @@ def media_upgrade(registered)
             "using the previous product. The packages from the registration " \
             "repositories can conflict with the new packages.</p>")
       end
+
+      def upgraded_base_product
+        # temporarily run the update mode to let the solver select the product for upgrade
+        # (this will correctly handle possible product renames)
+        Yast::Pkg.PkgUpdateAll({})
+        product = Y2Packager::Product.selected_base
+
+        # restore the initial status, the package update will be turned on later again
+        Yast::Pkg.PkgReset
+        changed = Yast::Pkg.ResolvableProperties("", :package, "").select do |p|
```

Comment:
Today we kill another instance of this problematic call and now we reintroducing it again. I worry it again kill memory requirements.

Reply:
From those two use cases it looks like we need new call like "PackagesByStatus" that accept one or more statuses adn return package names for that.